install/kubernetes/helm/istio/templates/sidecar-injector-configmap.yaml

```diff
@@ -138,7 +138,7 @@ data:
         - {{ "[[ .ProxyConfig.ProxyAdminPort ]]" }}
         {{ "[[ if gt .ProxyConfig.Concurrency 0 -]]" }}
         - --concurrency
-        - {{ "[[ .ProxyConfig.Concurrency ]]" }}
+        - {{ "[[ annotation .ObjectMeta `sidecar.istio.io/concurrency` .ProxyConfig.Concurrency ]]" }}
```

Comment:
This will only work if concurrency is not set to `0`

If annotation is set, then it should be used regardless of the global value.

Reply:
Yes, but that becomes a riskier change - and probably a much larger one ( the annotation doesn't have any gt 0). 

How about we document that if Concurrency is set to 0 in global settings it can't be overriden, and we fix it in 1.2 if anyone complains ? It's better than nothing, and the risks are not worth it.

Reply:
For every new Istio annotation we SHOULD provide a `const` in a Golang source file with a comment explaining the annotation.

See https://github.com/istio/istio.io/issues/3191 for rationale.

Reply:
agree.. no more obscure annotations.. Also, where is the documentation stating that if concurrency is 0, you can't override it?

Reply:
Ok, closing the PR in 1.1, we'll do it in master. 

Annotations and values.yaml need to be documented for sure - but I'm not sure there is any rule that all settings SHOULD have a const in Golang source file, in particular if they are not used in any way in the golang source.

Reply:
We can at least document that users can edit the inject template manually if they run into problems, and we'll have a fix allowing pods to override in future.

Reply:
And regarding 'obscure' - the annotations have the same name as the setting in Values.yaml, it's a pattern we've discussed and seems to have agreed on ( that we'll keep the names in sync, and document the values that can be overriden by annotations in the place where the main setting is defined - proto or values.yaml ).

That seems the proper place to document - since users don't read the golang source code, and 
they match the config they use to set the defaults..

Reply:
Although users don't read the source code when I attempted to create a "master glossary of annotations" it was suggested that I should close it and we would generate the list based on the source code similar to what we do for the protobufs.  https://github.com/istio/istio.io/issues/3191